File: BAEKJOON/Python/16000/16800/16890.py
"""
Title : 창업
Link : https://www.acmicpc.net/problem/16890
"""

from collections import deque
import sys

input = sys.stdin.readline

if __name__ == "__main__":
    ko_alp = sorted(list(input().strip()))
    cu_alp = sorted(list(input().strip()), reverse=True)
    L = len(ko_alp)

    ko, cu = deque(ko_alp[: L - L // 2]), deque(cu_alp[: L // 2])
    ans = [""] * L
    l_idx, r_idx = 0, L - 1

    for t in range(L - 1):
        if ko[0] >= cu[0]:
            if t % 2:
                ans[r_idx] = cu.pop()
            else:
                ans[r_idx] = ko.pop()
            r_idx -= 1
        else:
            if t % 2:
                ans[l_idx] = cu.popleft()
            else:
                ans[l_idx] = ko.popleft()
            l_idx += 1
    if ko:
        ans[l_idx] = ko[0]
    else:
        ans[l_idx] = cu[0]
    print("".join(ans))

File: BAEKJOON/Javascript/16000/16800/16890.js
/*
Title : 창업
Link : https://www.acmicpc.net/problem/16890
*/

const fs = require("fs");
const filepath =
  process.platform === "linux" ? "/dev/stdin" : __dirname + "/input.txt";
const input = fs.readFileSync(filepath).toString().trim().split("\n");

let ko_alp = [];
let cu_alp = [];

for (let i = 0; i < input[0].length; i++) {
  ko_alp.push(input[0][i]);
  cu_alp.push(input[1][i]);
}
const L = ko_alp.length;

ko_alp.sort((a, b) => a.charCodeAt(0) - b.charCodeAt(0));
cu_alp.sort((a, b) => b.charCodeAt(0) - a.charCodeAt(0));

let ko = [];
let cu = [];
for (let i = 0; i < parseInt(L / 2); i++) {
  ko.push(ko_alp[i]);
  cu.push(cu_alp[i]);
}
if (L % 2 === 1) {
  ko.push(ko_alp[L - parseInt(L / 2)]);
}

let ans = new Array(L);
let [lIdx, rIdx] = [0, L - 1];

for (let t = 0; t < L - 1; t++) {}

let tmp = "";
console.log();
